### PR TITLE
[move] fix resource account address derivation

### PR DESCRIPTION
**Pull request by alinush**
### Description

Take it away @movekevin.

Will update the description soon.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4787)

### CODE REVIEW

File: types/src/account_address.rs [hunk 1]
```diff
@@ -164,6 +164,7 @@ pub fn create_stake_pool_address(
     full_seed.extend(bcs::to_bytes(&operator).unwrap());
     full_seed.extend(SALT);
     full_seed.extend(seed);
+    full_seed.push(255);
```
Comment (davidiw) [inline types/src/account_address.rs:167]: This is redundant

File: types/src/account_address.rs [hunk 2]
```diff
@@ -177,6 +178,7 @@ pub fn create_vesting_contract_address(
     full_seed.extend(bcs::to_bytes(&nonce).unwrap());
     full_seed.extend(VESTING_POOL_SALT);
     full_seed.extend(seed);
+    full_seed.push(255);
```
Comment (davidiw) [inline types/src/account_address.rs:181]: This is redundant

**Review by davidiw — COMMENTED** [review]
Look at account_address.rs

File: types/src/account_address.rs (hunk 1 above)
Comment (movekevin) [reply types/src/account_address.rs:167]: Why is this redundant? Without this, the tests fail.

File: types/src/account_address.rs (hunk 2 above)
Comment (movekevin) [reply types/src/account_address.rs:181]: ditto

Comment (davidiw) [reply types/src/account_address.rs:181]: Reviewing on phone is a bad idea

**Review by davidiw — APPROVED** [review]

**Review by movekevin — APPROVED** [review]

File: types/src/account_address.rs [hunk 3]
```diff
@@ -193,6 +193,7 @@ pub fn create_vesting_pool_address(
 pub fn create_resource_address(address: AccountAddress, seed: &[u8]) -> AccountAddress {
     let mut input = bcs::to_bytes(&address).unwrap();
     input.extend(seed);
+    input.push(255);
```
Comment (alinush) [inline types/src/account_address.rs:196]: @movekevin, can you use the`DERIVE_RESOURCE_ACCOUNT_SCHEME` constant here?

**Pull request merged**